### PR TITLE
Implement getTotalChainConfigurations

### PR DESCRIPTION
**Pull request by RayXpub**
## Motivation
The goal of this PR is to add a getter function to the `CCIPConfig` contract to fetch the total number of chains configured. This value is necessary to correctly call `getAllChainConfigs` and  previously, the only way to access it was through events.

## Solution
Implement `getTotalChainConfigurations`.

### CODE REVIEW

File: contracts/src/v0.8/ccip/capability/CCIPConfig.sol [hunk 1]
```diff
@@ -101,6 +101,12 @@ contract CCIPConfig is ITypeAndVersion, ICapabilityConfiguration, OwnerIsCreator
     return i_capabilitiesRegistry;
   }
 
+  /// @notice Returns the total number of chains configured.
+  /// @return The total number of chains configured.
+  function getTotalChainConfigurations() external view returns (uint256) {
```
Comment (RyanRHall) [inline contracts/src/v0.8/ccip/capability/CCIPConfig.sol]: nit: I usually go with "num" or "length" when talking about an amount ex `getNumChainConfigurations()`. "total" feels like a "sum" ex getTotalAmountOwed() is like "get the sum of my debts"

**Review by RyanRHall — APPROVED** [review]

File: contracts/src/v0.8/ccip/capability/CCIPConfig.sol (hunk 1 above)
Comment (RayXpub) [reply contracts/src/v0.8/ccip/capability/CCIPConfig.sol]: Renamed

Comment (makramkd) [reply contracts/src/v0.8/ccip/capability/CCIPConfig.sol]: @RayXpub did you push the rename? Still see it as "total"

**Review by makramkd — APPROVED** [review]

File: contracts/src/v0.8/ccip/capability/CCIPConfig.sol (hunk 1 above)
Comment (makramkd) [reply contracts/src/v0.8/ccip/capability/CCIPConfig.sol]: nvm, github mobile gave me a stale view of the diff

**Review by elatoskinas — APPROVED** [review]

**Pull request merged**